apps/vaporgui/TFWidget.cpp

```diff
@@ -133,69 +142,94 @@ void TFWidget::Reinit(TFFlags flags) {
 }
 
 TFWidget::~TFWidget() {
-	if (_minCombo) {
+	if (_minCombo != nullptr) {
 		delete _minCombo;
-		_minCombo = NULL;
+		_minCombo = nullptr;
 	}   
-	if (_maxCombo) {
+	if (_maxCombo != nullptr) {
 		delete _maxCombo;
-		_maxCombo = NULL;
+		_maxCombo = nullptr;
 	}   
-	if (_rangeCombo) {
+	if (_rangeCombo != nullptr) {
 		delete _rangeCombo;
-		_rangeCombo = NULL;
+		_rangeCombo = nullptr;
 	}   
+    
+    if (_loadTFDialog != nullptr) {
+        delete _loadTFDialog;
+        _loadTFDialog = nullptr;
+    }
 }
 
 void TFWidget::loadTF() {
 	string varname = _rParams->GetColorMapVariableName();
-	if (varname.empty()) return;
-
-	SettingsParams* sP;
-	sP = (SettingsParams*)_paramsMgr->GetParams(SettingsParams::GetClassType());
-	string path = sP->GetTFDir();
-
-	fileLoadTF(varname, path.c_str(), true);
-}
+	if (varname.empty()) 
+        return;
 
-void TFWidget::fileLoadTF(
-	string varname, const char* startPath, bool savePath
-) {
-	QString s = QFileDialog::getOpenFileName(0,
-					"Choose a transfer function file to open",
-					startPath,
-					"Vapor 3 Transfer Functions (*.tf3)");
+    int rc = _loadTFDialog->exec();
+    if (rc == CANCEL)
+        return; // cancel event
 
-	// Null string indicates nothing selected
-	if (s.length()==0) return;
-	else {
-		SettingsParams* sP;
-		sP = (SettingsParams*)_paramsMgr->GetParams(SettingsParams::GetClassType());
-		sP->SetTFDir(s.toStdString());
-	}
+    string fileName = _loadTFDialog->GetSelectedFile();
+    if ( !selectedTFFileOk( fileName ) )
+        return;
 
-	// Force name to end with .tf3
-	if (!s.endsWith(".tf3")) {
-		s += ".tf3";
-	}
+	SettingsParams* sP;
+	sP = (SettingsParams*)_paramsMgr->GetParams(SettingsParams::GetClassType());
+	sP->SetTFDir(fileName);
 
 	MapperFunction *tf = _rParams->GetMapperFunc(varname);
 	assert(tf);
-    
-    vector<double> defaultRange;
-    _dataMgr->GetDataRange(0, varname, 0, 0, 1, defaultRange);
-
-    cout << "defaultRange " << defaultRange[0] << " " << defaultRange[1] << endl;
+    float cachedMin = tf->getMinMapValue();
+    float cachedMax = tf->getMaxMapValue();
+    int numOpacityMaps = tf->getNumOpacityMaps();
+    std::vector<std::vector<double>> controlPoints;
+    for (int i=0; i<numOpacityMaps; i++) {
+        controlPoints.push_back(tf->GetOpacityMap(i)->GetControlPoints());
+    }
 
-	int rc = tf->LoadFromFile(s.toStdString(), defaultRange);
+	_paramsMgr->BeginSaveStateGroup("Loading Transfer Function from file");
+ 
+	rc = tf->LoadFromFile(fileName);
 	if (rc<0) {
 		MSG_ERR("Error loading transfer function");
```

Comment:
You are returning from within a "StateSaveGroup". You need to call paramsMgr->EndSaveStateGroup() or you'll mess up the session state